src/utils/with-react.js

```diff
@@ -73,4 +79,61 @@ const applyDefaults = ceInst => {
 
 export { applyDefaults };
 
-export default createElement => val(createElement);
+const distributeProperties = ({ children, ...properties }, componentClass) => {
+  // initialize
+  const attrs = {};
+  const props = {};
+  let map;
+  // iterate over all properties
+  Object.keys(properties).forEach(name => {
+    let value = properties[name];
+    // classify property by type to select correct map object
+    // (note that unregistered properties are classified as attr(ibute)s via their undefined .type)
+    const event = name.indexOf('on') === 0 && Function; // Function if onXXX event name, false otherwise
+    const className = name === 'className' && name;
+    const defaultType = { type: event };
+    const declaredType = (componentClass.properties[name] || defaultType).type;
+    const type = event || className || declaredType;
```

Comment:
@markus-walther please rewrite this in a more readable way. Too much logic on one line and too much underneath JS know how is required. Also relying so hard on falsy / truly can be risky and hard to debug if written like this.

Reply:
I honestly think is sexy like this, but im pretty sure that is not very readable for everyone. I will ask for other peoples opinion

Reply:
done, rewritten via if's